### DISCUSSION TITLE
Does EC affect upload speed?

### DISCUSSION THREAD

**Discussion by Clara12062** [minio/minio#18560] [Q&A] [answered]
Hi,

&emsp; Without considering high availability, I want to know what settings, apart from hardware resources such as CPU, disk type, and network, can affect upload speed. For example, if I upload a 10M object and change the number of EC parity disks(`mc admin config set ALIAS storage_class standard=EC:1`), will the upload speed be affected?

&emsp; *I deployed minio with 3 nodes, each with 4 drivers(total 12 drivers).* I used warp for testing, uploading a 1M size object(`warp put --host=IP:PORT --bucket=test --access-key=minioUser --secret-key=minioPwd --obj.size=1M --concurrent=500 --benchdata=test --duration=1m`), and found that the change was not significant.

![GroupColumn-20231129](https://github.com/minio/minio/assets/111720237/ddad62eb-d87a-4393-b8f7-62aeda1dfa60)

&emsp; My guess is that when uploading objects(>1M), as the number of EC parity disks decreases and the number of EC data disks increases, the size of each shard decreases and the upload speed increases as a result. Is that right?

**Comment by harshavardhana** [comment, marked as answer]
Correct, parity I/O is what you loose for writes. But thats what it means to have tolerance. You need for pay for HA cost during writes since to know your content is intact you need some form of encoding scheme in this case it's erasure coding.

If you go with zero parity writes you will basically get maximum possible performance but with no HA, you lose a disk then your data is unrecoverable.

**Reply by klauspost** [reply]
Also with 1MB objects you are much more likely to be IOPS limited. In that case the slightly smaller amount of data `EC:0` will be writing will not really make much of a difference.

**Reply by Clara12062** [reply]
I see
That explains my test performance.
Thank for your reply! :smile:
